Sources/PayPalWebCheckout/PayPalWebCheckoutClient.swift

```diff
@@ -101,3 +96,19 @@ public class PayPalWebCheckoutClient {
         delegate?.payPalDidCancel(self)
     }
 }
+
+// MARK: - ASWebAuthenticationPresentationContextProviding conformance
+
+extension PayPalWebCheckoutClient: ASWebAuthenticationPresentationContextProviding {
+    
+    public func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        if #available(iOS 15, *) {
+            let firstScene = UIApplication.shared.connectedScenes.first as? UIWindowScene
+            let window = firstScene?.windows.first { $0.isKeyWindow }
+            return window ?? ASPresentationAnchor()
+        } else {
+            let window = UIApplication.shared.windows.first { $0.isKeyWindow }
+            return window ?? ASPresentationAnchor()
+        }
```

Comment:
Could we extract this into a default implementation somewhere to avoid code duplication?

Reply:
Great callout! 

So I can't do a protocol extension on `ASWebAuthenticationPresentationContextProviding` because it inherits from `NSObject` ([see Apple docs](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationpresentationcontextproviding?language=objc)). Also, even if we could do it, I'm not sure we'd want to because then merchants using `ASWebAuthenticationPresentationContextProviding` in other ways in their app would be stuck with whatever default implementation we need to give it for our SDK.

In terms of trying to re-use (instead of duplicate) this code, I think we'd have to do inheritance or move the logic inside of the `presentationAnchor` into some global/shared func across modules. Since `CardClient` and `PayPalWebCheckoutClient` are merchant integration points, the parent class would need to be public and may cause more confusion to the merchant 🤔.

We could experiment with the later option though. Maybe this can be a new ticket / punt 🏈 ? Also, the `PaymentFlow` module on BT exists to solve this kind of problem. Not needing to redo the logic of switching to a WebView per payment method. Not sure if that is really worthwhile here, but just calling it out.

If you have other ideas please let me know too!